cpp/examples/README.md

```diff
@@ -0,0 +1,43 @@
+# libcugraph examples
+
+Example codes on how to use libcugraph.
+
+## Contents
+
+- simple
+
+  - single_gpu_application: example code on how to use libgraph to run different graph algorithms on a single-GPU node.
+
+  - multi_gpu_application: example code on how to use libgraph to run different graph algorithms on a multi-GPU node.
+
+- advanced
+  -  graph_partition: code to explain vertex and edge partitioning in cugraph.
+
+  -  cugraph_operations: example code for using cugraph primitives for simple graph operations needed to implement graph algorithms. 
+
+## Build instructions:
+
+Run `build.sh` to from examples dir to build the above listed examples.
+
+```sh
+~/cugraph/cpp/examples$./build.sh
+```
+
+## Run instructions
+
+For single_gpu
+
+`path_to_executable path_to_a_csv_graph_file [memory allocation mode]`
```

Comment:
As we are only focusing on example rather than real application, removed the memory allocation mode parameter and used rmm::mr::cuda_memory_resource for all the examples.